python/lsst/analysis/drp/makeQaTractTables.py

```diff
@@ -0,0 +1,418 @@
+
+
+class MakeForcedQaTractTablesTaskConnections(BaseMakeQaTractTablesTaskConnections,
+                                             defaultTemplates={"coaddName": "deep"},
+                                             dimensions=("tract", "skymap")):
+    pass
+
+
+class MakeForcedQaTractTablesTaskConfig(BaseMakeQaTractTablesTaskConfig,
+                                        pipelineConnections=MakeForcedQaTractTablesTaskConnections):
+    # We want the following to come from the *_meas catalogs as they reflect
+    # what happened in SFP calibration.
+    columnsToCopyFromMeas = pexConfig.ListField(
+        dtype=str,
+        default=["calib_", ],
+        doc=("List of string \"prefixes\" to identify the columns to copy.  All columns with names "
+             "that start with one of these strings will be copied from the *_meas catalogs into the "
+             "*_forced_src catalogs UNLESS the full column name contains one of the strings listed "
+             "in the notInColumnStrList config.")
+    )
+    # We want the following to come from the *_ref catalogs as they reflect
+    # the forced measurement states.
+    columnsToCopyFromRef = pexConfig.ListField(
+        dtype=str,
+        default=["detect_", "merge_peak_sky", "merge_measurement_", ],
+        doc=("List of string \"prefixes\" to identify the columns to copy.  All columns with names "
+             "that start with one of these strings will be copied from the *_ref catalogs into the "
+             "*_forced_src catalogs UNLESS the full column name contains one of the strings listed "
+             "in the notInColumnStrList config.")
+    )
+
+
+class MakeForcedQaTractTablesTask(BaseMakeQaTractTablesTask):
+    """Read in {coaddName}Coadd_obj tables and aggregate QA columns by tract.
+
+    The values of interest here are those from the forced catalogs (i.e.
+    "forced_src") and, in this case, certain columns from the "meas" and "ref"
+    catalogs that don't get propagated to the "forced_src" catalogs are added
+    to the persisted table, which has dataset type "qaTractTable_forced".
+
+    See BaseMakeQaTractTablesTask for full description.
+    """
+
+    ConfigClass = MakeForcedQaTractTablesTaskConfig
+    _DefaultName = "makeForcedQaTractTablesTask"
+
+    colsToLoadList = None
+    refColsToLoadList = None
+    measColsToLoadList = None
+
+    def collateTable(self, inputCat, inputCatCols, activeFilterName, filterName, haveFilter):
+        """Implementation of BaseMakeQaTractTablesTask.collateTable for forced
+        measurements.
+        """
+        measTypeStr = "forced_src"
+        if self.colsToLoadList is None:
+            forcedColsList = [tup[2] for tup in inputCatCols if tup[0] == measTypeStr
+                              and tup[1] == activeFilterName]
+            self.colsToLoadList = deriveColumnsList(forcedColsList, self.config.baseColStrList,
+                                                    self.config.notInColStrList)
```

Comment:
It looks to me like the list comprehension that happens before a call to `deriveColumnsList` is always the same.  Would it make sense to move it inside `deriveColumnsList`, and make `deriveColumnsList` take `(measTypeStr, inputCatCols, activeFilterName)` instead of its current first argument?

Reply:
I also added an allowance for just a `list` of `str` being provided.